src/main/kotlin/org/rust/ide/refactoring/changeSignature/RsChangeSignatureHandler.kt

```diff
@@ -0,0 +1,342 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.refactoring.changeSignature
+
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.refactoring.RefactoringFactory
+import com.intellij.refactoring.changeSignature.ChangeInfo
+import com.intellij.refactoring.changeSignature.ChangeSignatureHandler
+import com.intellij.refactoring.changeSignature.ChangeSignatureProcessorBase
+import com.intellij.usageView.BaseUsageViewDescriptor
+import com.intellij.usageView.UsageInfo
+import com.intellij.usageView.UsageViewDescriptor
+import org.rust.ide.inspections.fixes.deleteElementWithCommasAndWhitespace
+import org.rust.ide.presentation.renderInsertionSafe
+import org.rust.ide.refactoring.findBinding
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.RsElementTypes.COMMA
+import org.rust.lang.core.psi.ext.*
+import org.rust.lang.core.types.ty.TyUnit
+import org.rust.lang.core.types.type
+import org.rust.openapiext.elementUnderCaretInEditor
+import org.rust.stdext.mapToMutableList
+
+class RsChangeSignatureHandler : ChangeSignatureHandler {
+    override fun getTargetNotFoundMessage(): String = "Must target function"
+
+    override fun findTargetMember(element: PsiElement): PsiElement? = element.ancestorOrSelf() as? RsFunction
+
+    override fun invoke(project: Project, elements: Array<out PsiElement>, dataContext: DataContext?) {
+        val function = elements.firstOrNull() as? RsFunction ?: return
+        showRefactoringDialog(function)
+    }
+
+    override fun invoke(project: Project, editor: Editor?, file: PsiFile?, dataContext: DataContext?) {
+        val function = dataContext?.elementUnderCaretInEditor as? RsFunction ?: return
+        showRefactoringDialog(function)
+    }
+
+    private fun showRefactoringDialog(function: RsFunction) {
+        val config = RsFunctionSignatureConfig.create(function)
+        val project = function.project
+
+        showChangeFunctionSignatureDialog(project, config)
+    }
+}
+
+class RsChangeSignatureProcessor(project: Project, changeInfo: ChangeInfo)
+    : ChangeSignatureProcessorBase(project, changeInfo) {
+    override fun performRefactoring(usages: Array<out UsageInfo>?) {
```

Comment:
I implemented the `findUsages` method and it seems that it is enough to display preview.

Reply:
Looks like [all other languages implement `ChangeSignatureUsageProcessor`](https://user-images.githubusercontent.com/6505554/104351149-15e67b80-5516-11eb-9d42-e1454599d871.png), but I don't know whether or why it is needed.

Btw if I change function name (using preview to show usages) and then undo change signature refactoring, then dialog without function name can appear:
![image](https://user-images.githubusercontent.com/6505554/104351505-88575b80-5516-11eb-90ce-f67a31ae2f50.png)

Reply:
Ok, I'll implement it.

Reply:
I implemented it, but the undo button still says `Changing signature of ?`. I couldn't find from where the error message comes, any ideas?

Reply:
Oh, actually it is because [`RsFindUsagesProvider#getDescriptiveName`](https://github.com/intellij-rust/intellij-rust/blob/b0be6aaeff0295cda97016507a0bb642d76b11a5/src/main/kotlin/org/rust/ide/search/RsFindUsagesProvider.kt#L23) always returns empty string